src/stuf/stuf_zds/client.py

```diff
@@ -88,142 +82,136 @@ def xml_value(xml, xpath, namespaces=nsmap):
+        Initialize the client instance.
+
+        :arg options: Values from the ``ZaakOptionsSerializer``, amended with
+          'omschrijving' and 'referentienummer'.
         """
-        self.service = service
+        super().__init__(
+            service,
+            request_log_hook=logevent.stuf_zds_request,
+        )
         self.options = options
 
-        self._global_config = GlobalConfiguration.get_solo()
-
-    def _get_request_base_context(self):
-        return {
-            "zender_organisatie": self.service.zender_organisatie,
-            "zender_applicatie": self.service.zender_applicatie,
-            "zender_gebruiker": self.service.zender_gebruiker,
-            "zender_administratie": self.service.zender_administratie,
-            "ontvanger_organisatie": self.service.ontvanger_organisatie,
-            "ontvanger_applicatie": self.service.ontvanger_applicatie,
-            "ontvanger_gebruiker": self.service.ontvanger_gebruiker,
-            "ontvanger_administratie": self.service.ontvanger_administratie,
-            "tijdstip_bericht": fmt_soap_datetime(timezone.now()),
-            "tijdstip_registratie": fmt_soap_datetime(timezone.now()),
-            "datum_vandaag": fmt_soap_date(timezone.now()),
-            "gemeentecode": self.options["gemeentecode"],
-            "zds_zaaktype_code": self.options["zds_zaaktype_code"],
-            "zds_zaaktype_omschrijving": self.options.get("zds_zaaktype_omschrijving"),
-            "zds_zaaktype_status_code": self.options.get("zds_zaaktype_status_code"),
-            "zds_zaaktype_status_omschrijving": self.options.get(
-                "zds_zaaktype_status_omschrijving"
-            ),
-            "zaak_omschrijving": self.options["omschrijving"],
-            "zds_documenttype_omschrijving_inzending": self.options[
-                "zds_documenttype_omschrijving_inzending"
-            ],
-            "zds_zaakdoc_vertrouwelijkheid": self.options[
-                "zds_zaakdoc_vertrouwelijkheid"
-            ],
-            "referentienummer": str(uuid4()),
-            "global_config": self._global_config,
-        }
-
-    def _wrap_soap_envelope(self, xml_str: str) -> str:
-        return loader.render_to_string(
-            "stuf_zds/soap/includes/envelope.xml",
+    def build_base_context(self) -> dict[str, Any]:
+        context = super().build_base_context()
+        now = timezone.now()
+        context.update(
             {
-                "soap_version": self.service.soap_service.soap_version,
-                "soap_use_wss": (
-                    self.service.soap_service.endpoint_security
-                    in [EndpointSecurity.wss, EndpointSecurity.wss_basicauth]
+                "tijdstip_registratie": fmt_soap_datetime(now),
+                "datum_vandaag": fmt_soap_date(now),
+                "gemeentecode": self.options.get("gemeentecode", ""),
```

Comment:
there are a lot of tests that don't properly pass in these options :(

I'm also wondering if this should even be part of the base context... The type annotation now at least reports these errors.